ksqldb-parser/src/main/java/io/confluent/ksql/util/QueryMetaDataCollector.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.util;
+
+import com.google.common.collect.ImmutableMap;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import org.joda.time.DateTime;
+
+public final class QueryMetaDataCollector {
+
+  private QueryMetaDataCollector() {}
+
+  public static ImmutableMap<String, String> buildMetaData(
+      final KsqlConfig config,
```

Comment:
First arg should be  `String namespace` rather than KsqlConfig so that you don't tight couple this with configuration stack